Feature/fixed histograms

jonathanbostock opened the pull request:
Fixed the histograms, made some minor refactoring changes, and got the pre-commit hook running.

jonathanbostock committed:
Histograms now work sensibly (using seaborn). Moved the logprob code to
model_calling and improved it by making it async.

jonathanbostock committed:
Refactored the modules in utils to make them a bit more understandable.
We should really rename the whole project to have fewer characters I
think.

jonathanbostock committed:
GIT PRE-COMMIT NOW WORKS HOLY HELL

NelsonG-C commented on src/lasr_labs_2025_control_project/scripts/logprob_analysis.py:
```diff
@@ -1,38 +1,56 @@
+            logprob_calculator=calculate_logprobs,
         )
 
         config = build_optimization_config(
```
I think the pass can be removed on line 68

jonathanbostock replied:
Done

NelsonG-C commented on src/lasr_labs_2025_control_project/scripts/logprob_analysis.py:
```diff
@@ -52,7 +70,7 @@ def main():
 
 def _get_logprobs_from_dataset(
```
The extract code function here uses the <final_code></final_code> match, but the true final code is what is submitted by the submit tool - though I believe they shouldn't be any different. If we confirm no difference then this will be fine

jonathanbostock replied:
This pulls from the messages rather than finding the tool call. I can't find where tool calls are actually being saved (it's not in the "tool_calls" column of the dataframe for a start). Not sure what's going on here exactly, on manual inspection it's definitely where the model has written its code, but I don't know if this will continue to work when there's an agent loop going on.

NelsonG-C commented on src/lasr_labs_2025_control_project/utils/model_calling/logprobs.py:
Okay so we run this instead of the monitor in the pipeline. To confirm, are we doing it like this because control_arena / Inspect doesn't allow us to get the log probs from the model in the existing flow?

NelsonG-C replied:
I've run out of time to review all of the build_logprob_caller atm and want to get comments through so FYI on that

jonathanbostock replied:
Yeah we run this instead of the monitor. I can look into getting logprobs in control_arena if we want, but what we have now seems like the clear MVP solution.

NelsonG-C commented on src/lasr_labs_2025_control_project/utils/numerical_pipeline/visualization.py:
Looks good to me

NelsonG-C commented on src/lasr_labs_2025_control_project/utils/model_calling/logprobs.py:
```diff
@@ -0,0 +1,223 @@
+)
+
+# Default prompts used for logprob evaluation tasks.
+LOGPROBS_SYSTEM_PROMPT = (
```
If these will be iterated on we should extract these out of these out of this file into a prompts file

jonathanbostock replied:
Split into a .py file for now.

NelsonG-C commented on src/lasr_labs_2025_control_project/utils/model_calling/logprobs.py:
```diff
@@ -0,0 +1,223 @@
+
+    # Find NO
+    no_candidates = [lp for lp in top if getattr(lp, "token", "").lower() == "no"]
+    no_lp = no_candidates[0] if no_candidates else top[-1]
```
I think i'm confused on what top is exactly here and what top[-1] would be returning if it finds no candidates. Can you explain to me / add a comment here so this is clearer

jonathanbostock replied:
Have removed this top[-1] nonsense, just replaced with returning None if no candidates found. Code runs the same on existing data.

jonathanbostock committed:
Minor changes to make some files more comprehensible (and fail less
weirdly) for PR.

jonathanbostock committed:
Changed the logprob_analysis script to save to csv then load the csv,
and to filter the samples. Filtering gives us really low success rates
but at least our data is clean.

jonathanbostock committed:
Slightly changed how paraphrasers are handled in utils. Now they just
operate on list[str] which is much more sane than the object stuff.
Could maybe keep a version of the object stuff when merging.

NelsonG-C commented on src/lasr_labs_2025_control_project/scripts/logprob_analysis.py:
```diff
@@ -0,0 +1,269 @@
+
+    for model_name in models.keys():
+        self_data = pd.read_csv(
+            Path(f"logs/4.1-4o/4.1-attacks-{model_name}-logprobs-summarizer.csv")
```
We're passing these into both functions, should just extract out as param into main() and pass through

NelsonG-C commented on src/lasr_labs_2025_control_project/utils/paraphrasers/__init__.py:
I'll take this out of the init

NelsonG-C committed:
remove paraphrasers from the init file

NelsonG-C committed:
extract out file pathing naming to main function for easy use

jonathanbostock committed:
Merge branch 'main' into feature/fixed-histograms
Had to discard a bunch of changes to data_generation.py since they seem
to be broken.
